.circleci/config.yml

```diff
@@ -229,6 +229,26 @@ references:
         }'
 
 jobs:
+  validate_version:
+    docker:
+      - image: circleci/node:12.16.2-browsers
+    environment:
+      VERSION: << pipeline.git.tag >>
+    steps:
+      - when:
+          condition: << pipeline.git.tag >>
+          steps:
+            - checkout
+            - run:
+                name: Ensure package.json Version And Tag Match
+                command: node ./validate_tag.js $VERSION
+      - unless:
+          condition: << pipeline.git.tag >>
+          steps:
+            - run:
+                name: Skip Validation
+                command: echo "Not on a tag. Skipping version validation."
```

Comment:
This new step runs before the `build` one and if the build started with a tag, checks that the version in package.json matches the tag.

I'm not happy that we now have to run this step on every build, but at the same time it adds less that 15s to a build that is in the order of minutes. It seems like a reasonable tradeoff to prevent future error tagging new versions.

I thought about adding the condition in the `wp-desktop` workflow, but it already uses one and from [the docs](https://circleci.com/docs/reference-2-1/#the-when-attribute) it doesn't look like you can nest or compose them.